File: bin/w2r.js
#!/usr/bin/env node

const { spawnSync } = require("node:child_process");
const path = require("node:path");

const cliPath = path.resolve(__dirname, "..", "src", "w2r", "cli.py");
const args = [cliPath, ...process.argv.slice(2)];

function runWith(pythonBin) {
  return spawnSync(pythonBin, args, {
    stdio: "inherit",
    env: process.env
  });
}

let result = runWith("python3");
if (result.error) {
  result = runWith("python");
}

if (result.error) {
  console.error("Error: Python 3.10+ is required to run w2r.");
  process.exit(2);
}

process.exit(result.status ?? 1);

File: w2r.py
#!/usr/bin/env python3
"""Thin compatibility shim for running w2r from source checkout."""
from pathlib import Path
import sys

ROOT = Path(__file__).resolve().parent
SRC = ROOT / "src"
if str(SRC) not in sys.path:
    sys.path.insert(0, str(SRC))

from w2r.cli import main


if __name__ == "__main__":
    raise SystemExit(main())
